src/reps/rep-utils.js

```diff
@@ -373,6 +376,25 @@ function getGripPreviewItems(grip) {
   return [];
 }
 
+function safeObjectLink(props, config, ...children) {
+  const {
+    objectLink,
+    object,
+  } = props;
+
+  if (objectLink) {
+    return objectLink(Object.assign({
+      object
+    }, config), ...children);
+  }
+
+  if (Object.keys(config).length === 0 && children.length === 1) {
+    return children[0];
+  }
+
+  return React.DOM.span(config, ...children);
+}
+
 module.exports = {
   createFactories,
```

Comment:
I would say :  
- create a bug in mc to duplicate createFactories from reps in a mc util and use it everywhere
- update the image in circle after bug has been resolved
- nuke createFactories here